tracing/tracer.go

```diff
@@ -0,0 +1,75 @@
+package tracing
+
+import (
+	"io"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	jeagerConf "github.com/uber/jaeger-client-go/config"
+)
+
+type Tracer struct {
+	ServiceName  string
+	Provider     string
+	Logger       *logrus.Logger
+	JaegerConfig *JaegerConfig
+
+	tracer opentracing.Tracer
+	closer io.Closer
+}
+
+type JaegerConfig struct {
+	LocalAgentHostPort string
+	SamplerType        string
+	SamplerValue       float64
+	SamplerServerUrl   string
+}
+
+func (t *Tracer) Setup() {
+	switch t.Provider {
```

Comment:
I think it would be good to lowercase this so `JAEGER` and `jaeger` are equivalent

Reply:
Good idea 👍, will update